joblib/test/test_numpy_pickle.py

```diff
@@ -212,6 +249,46 @@ def test_memmap_persistence():
 
     nose.tools.assert_true(isinstance(b, np.memmap))
 
+    # Test with an object containing multiple numpy arrays
+    filename = env['filename'] + str(random.randint(0, 1000))
+    obj = ComplexTestObject()
+    numpy_pickle.dump(obj, filename)
+    obj_loaded = numpy_pickle.load(filename, mmap_mode='r')
+    nose.tools.assert_true(isinstance(obj_loaded, type(obj)))
+    nose.tools.assert_true(isinstance(obj_loaded.array_float, np.memmap))
+    nose.tools.assert_false(obj_loaded.array_float.flags.writeable)
+    nose.tools.assert_true(isinstance(obj_loaded.array_int, np.memmap))
+    nose.tools.assert_false(obj_loaded.array_int.flags.writeable)
+    # Memory map not allowed for numpy object arrays
+    nose.tools.assert_true(isinstance(obj_loaded.array_obj,
```

Comment:
This test would pass even if obj_loaded.array_obj was memmaped. since memmap is a subclass of ndarray.

Use something like:
```python
nose.tools.assert_false(isinstance(obj_loaded.array_obj, np.memmap))
```